### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by khushbuSatasiya**
Scrum master

### CODE REVIEW

File: .env [hunk 1]
```diff
@@ -0,0 +1,6 @@
+# REACT_APP_BASE_URL=https://montu-test.9span.in/api/v1
+REACT_APP_ENCRYPTION_KEY= ADFDasdafcsdfsdfsdfsd
+REACT_APP_BASE_URL=https://montu-test.9span.in/api/v1
+# REACT_APP_BASE_URL=http://192.168.1.124:3000/api/v1
```
Comment (daxesh-parmar-scaletech) [inline .env]: Remove commented code.

File: public/index.html [hunk 2]
```diff
@@ -22,7 +22,7 @@
       work correctly both with client-side routing and a non-root public URL.
       Learn how to configure a non-root public URL by running `npm run build`.
     -->
-  <title>Scrum master admin</title>
+  <title>Scrum master</title>
```
Comment (daxesh-parmar-scaletech) [inline public/index.html:26]: Add proper description related to app.

File: src/App.css [hunk 3]
```diff
@@ -36,3 +37,58 @@
...
+  font-size: 20px;
+}
+
+.rbc-month-view {
+  border-radius: 6px;
+}
+
+/* .rbc-button-link{
+  color:#3f4254 !important;
+} */
```
Comment (daxesh-parmar-scaletech) [inline src/App.css]: Remove commented all code from file.

File: src/App.tsx [hunk 4]
```diff
@@ -1,29 +1,35 @@
...
-      <Suspense
-        fallback={
-          <Group w="100%" h="500px" position="center" align="center">
-            <Loader size="xl" />
-          </Group>
-        }
-      >
-        <Routes>
-          <Route path="/user/list" element={<User />} />
+    <>
```
Comment (daxesh-parmar-scaletech) [inline src/App.tsx]: no need for fragment here

File: src/App.tsx [hunk 5]
```diff
@@ -1,29 +1,35 @@
...
+      <Layout>
+        <Suspense
+          fallback={
+            <Group w="100%" h="500px" position="center" align="center">
+              {/* <Loader size="xl" /> */}
```
Comment (daxesh-parmar-scaletech) [inline src/App.tsx]: why the loader is commented?

File: src/features/User/component/user.tsx [hunk 6]
```diff
@@ -1,137 +1,45 @@
...
-					handleEditPopup={handleEditPopup}
-					getUserList={getUserList}
-				/>
-			)}
-		</div>
-	);
+  const [searchValue, setSearchVal] = useDebouncedState("", 400, {
+    leading: true,
+  });
+  //   const selectedOrg = useSelector((state: State) => state?.auth?.selectedOrg);
```
Comment (daxesh-parmar-scaletech) [inline src/features/User/component/user.tsx]: remove commented code

File: src/features/User/component/viewUser.tsx [hunk 7]
```diff
@@ -1,68 +1,21 @@
...
-      vacationLeaves,
-      compensationLeaves,
-    });
-  };
-
-  const handleTimeSheetDetails = (workingHours) => {
-    setTotalWorkingHour(workingHours);
-  };
-
   const renderPaper = (label, value, color) => (
```
Comment (daxesh-parmar-scaletech) [inline src/features/User/component/viewUser.tsx:23]: Add constant array in `userConstants.tsx`

File: src/features/checkIn/component/checkIn.tsx [hunk 8]
```diff
@@ -0,0 +1,244 @@
...
+          setIsAlreadyCheckIn(true);
+        }
+        console.error(error);
+      }
+    },
+    [checkStatus]
+  );
+
+  /* add task button disable */
+  const isAddButtonDisabled = (employee: any) => {
```
Comment (daxesh-parmar-scaletech) [inline src/features/checkIn/component/checkIn.tsx]: give proper type. don't use any

File: src/features/checkOut/component/addExtraTaskForm.tsx [hunk 9]
```diff
@@ -0,0 +1,151 @@
...
+  Textarea,
+} from "@mantine/core";
+import { IconPlus, IconTrash } from "@tabler/icons-react";
+
+interface IProps {
+  form: any;
+  isShowForm: boolean;
+  setIsShowForm: (action: boolean) => void;
+  handleAddTaskBtn: () => void;
+  projects: any;
```
Comment (daxesh-parmar-scaletech) [inline src/features/checkOut/component/addExtraTaskForm.tsx]: don't use any

File: src/features/checkOut/component/addExtraTaskForm.tsx [hunk 10]
```diff
@@ -0,0 +1,151 @@
...
+  classes: any;
+}
+
+const AddExtraTaskForm: FC<IProps> = (props: IProps) => {
+  const { form, isShowForm, handleAddTaskBtn, projects, classes } = props;
+
+  /* add task button disable */
+  const isAddButtonDisabled = (employee: any) => {
```
Comment (daxesh-parmar-scaletech) [inline src/features/checkOut/component/addExtraTaskForm.tsx]: don't use any

File: src/features/checkOut/component/checkOut.tsx [hunk 11]
```diff
@@ -0,0 +1,242 @@
...
+import { API_CONFIG } from "shared/constants/api";
+
+import { IActionTime } from "features/dashboard/interface/dashboard";
+
+import CheckOutForm from "./checkOutForm";
+import { getProjectList } from "shared/util/utility";
+import CheckoutModals from "./checkoutModals";
+
+interface IProps {
+  enteredTask: any;
```
Comment (daxesh-parmar-scaletech) [inline src/features/checkOut/component/checkOut.tsx]: Remove any

File: src/features/checkOut/component/checkOut.tsx [hunk 12]
```diff
@@ -0,0 +1,242 @@
...
+  currentTime,
+  actionTime,
+}) => {
+  const [projects, setProjects] = useState([]);
+  const [isShowForm, setIsShowForm] = useState(false);
+  const [isLoading, setIsLoading] = useState(false);
+  const [isAlreadyCheckOut, setIsAlreadyCheckOut] = useState(false);
+  const [isConfirm, setIsConfirm] = useState(false);
+
+  const formatValues = (userTask: any) => {
```
Comment (daxesh-parmar-scaletech) [inline src/features/checkOut/component/checkOut.tsx:51]: optimize this function. put condition only in task array.

File: src/features/checkOut/component/checkOut.tsx [hunk 13]
```diff
@@ -0,0 +1,242 @@
...
+
+      <Modal
+        size="auto"
+        opened={isAlreadyCheckOut}
+        onClose={() => setIsAlreadyCheckOut(false)}
+        centered
+        padding={40}
+        radius="lg"
+        withCloseButton={false}
+      >
```
Comment (daxesh-parmar-scaletech) [inline src/features/checkOut/component/checkOut.tsx]: make separate component of modal

File: src/features/dashboard/component/dashboard.tsx [hunk 14]
```diff
@@ -0,0 +1,292 @@
...
+  const [isShowUserDetails, setIsShowUserDetails] = useState(false);
+  const [calendarIndicator, setCalendarIndicator] = useState([]);
+
+  /* API call for check status*/
+  const checkStatus = useCallback(async () => {
+    setIsActionLoader(true);
+    try {
+      await httpService.get(API_CONFIG.path.status).then((res) => {
+        setIsActionLoader(false);
+        setActionType(res.data.action);
```
Comment (daxesh-parmar-scaletech) [inline src/features/dashboard/component/dashboard.tsx]: set res.data.action in variable and use that in entire function.

File: src/features/dashboard/component/dashboard.tsx [hunk 15]
```diff
@@ -0,0 +1,292 @@
...
+    },
+    {
+      label: "Calendar",
+      value: "calendar",
+      content: <TeamCalendar />,
+    },
+  ];
+
+  return (
+    <>
```
Comment (daxesh-parmar-scaletech) [inline src/features/dashboard/component/dashboard.tsx]: No need of fragment here

File: src/features/dashboard/component/userDetail.tsx [hunk 16]
```diff
@@ -0,0 +1,365 @@
...
+    activeTab,
+    newToken,
+    totalWorkingHour,
+    leaveDetails,
+    setIsShowUserDetails,
+    isShowUserDetails,
+    calendarIndicator,
+  } = props;
+
+  const LEAVE_DETAILS = [
```
Comment (daxesh-parmar-scaletech) [inline src/features/dashboard/component/userDetail.tsx]: add LEAVE_DETAILS in a constant folder.

File: src/features/dashboard/component/userDetail.tsx [hunk 17]
```diff
@@ -0,0 +1,365 @@
...
+        <Button
+          compact
+          bg={"#eaf6ff"}
+          sx={{
+            "&:hover": {
+              backgroundColor: "#F1F1F2",
+            },
+          }}
+        >
+          <Anchor
```
Comment (daxesh-parmar-scaletech) [inline src/features/dashboard/component/userDetail.tsx]: use separate component to show anchor buttons.
and make it common as only links are changing.

File: src/features/dashboard/component/userInfoPopup.tsx [hunk 18]
```diff
@@ -0,0 +1,153 @@
...
+interface IProps {
+  isShowUserDetails: boolean;
+  setIsShowUserDetails: (action: boolean) => void;
+}
+
+const UserInfoPopup: FC<IProps> = ({
+  isShowUserDetails,
+  setIsShowUserDetails,
+}) => {
+  const [userDetail, setUserDetail] = useState<any>({});
```
Comment (daxesh-parmar-scaletech) [inline src/features/dashboard/component/userInfoPopup.tsx]: Remove any

File: src/features/leave/component/leave.tsx [hunk 19]
```diff
@@ -0,0 +1,191 @@
...
+    (
+      currentPage = pagination.currentPage,
+      recordPerPage = pagination.recordPerPage,
+      status?,
+      date?
+    ) => {
+      const params = {
+        currentPage,
+        recordPerPage,
+        status: status === undefined ? selectedStatus : status,
```
Comment (daxesh-parmar-scaletech) [inline src/features/leave/component/leave.tsx]: use this condition => `status ?? selectedStatus`

File: src/features/leave/component/leave.tsx [hunk 20]
```diff
@@ -0,0 +1,191 @@
...
+            label="Download Excel"
+            color="blue"
+            position="bottom"
+            withArrow
+          >
+            <ThemeIcon
+              w={40}
+              h={40}
+              onClick={getLeaveExcel}
+              // variant="light"
```
Comment (daxesh-parmar-scaletech) [inline src/features/leave/component/leave.tsx]: Remove commented code

File: src/features/project/components/teamCalendar.tsx [hunk 21]
```diff
@@ -0,0 +1,139 @@
...
+
+import { Calendar, momentLocalizer } from "react-big-calendar";
+import moment from "moment";
+
+import { DotIcon } from "shared/icons/icons";
+
+import "react-big-calendar/lib/css/react-big-calendar.css";
+
+interface IProps {
+  calendarInfo: any;
```
Comment (daxesh-parmar-scaletech) [inline src/features/project/components/teamCalendar.tsx]: Remove any

File: src/shared/components/noRecords/noRecords.tsx [hunk 22]
```diff
@@ -0,0 +1,44 @@
...
+      justify={"center"}
+    >
+      <Box
+        p={4}
+        mb={4}
+        sx={(theme) => ({
+          width: "48px",
+          fontSize: 0,
+          color:
+            theme.colorScheme === "dark"
```
Comment (daxesh-parmar-scaletech) [inline src/shared/components/noRecords/noRecords.tsx]: there is no need of dark theme condition

File: src/shared/components/notification/notification.tsx [hunk 23]
```diff
@@ -0,0 +1,53 @@
...
+      description: { color: theme.white },
+      closeButton: {
+        color: theme.white,
+        "&:hover": {
+          backgroundColor: theme.colors.blue[7],
+        },
+      },
+    }),
+  });
+};
```
Comment (daxesh-parmar-scaletech) [inline src/shared/components/notification/notification.tsx]: we can manage notifications in single component with props

File: src/shared/components/table/container/table.tsx [hunk 24]
```diff
@@ -1,99 +1,123 @@
...
+  return {
+    root: {
+      backgroundColor: "#fff",
+    },
+    header: {
+      backgroundColor: theme.colors.blue[5],
+      color: theme.white,
+      // '&& tr': { height: '50px', backgroundColor: theme.colors.blue[5] },
+      // '&& th': { fontSize: '16px', fontWeight: 'bold', color: theme.white },
+      // '&&& th:hover': { backgroundColor: theme.colors.blue[5], color: theme.white }
```
Comment (daxesh-parmar-scaletech) [inline src/shared/components/table/container/table.tsx]: remove commented code

File: src/shared/constants/validation-schema.ts [hunk 25]
```diff
@@ -0,0 +1,198 @@
...
+    ),
+    // otherwise: Yup.array().of(
+    //   Yup.array().of(
+    //     Yup.object().shape({
+    //       task: Yup.string(),
+    //       project: Yup.string(),
+    //       projectHours: Yup.string(),
+    //     })
+    //   )
+    // ),
```
Comment (daxesh-parmar-scaletech) [inline src/shared/constants/validation-schema.ts]: remove commented code

File: src/shared/constants/validation-schema.ts [hunk 26]
```diff
@@ -0,0 +1,198 @@
...
+  //       .test("project-required", "Project is required", function (value) {
+  //         const taskValue = this.parent.task;
+  //         if (taskValue === "") {
+  //           return true; // Project validation is skipped when task is empty
+  //         }
+
+  //         return !!value;
+  //       }),
+  //   })
+  // ),
```
Comment (daxesh-parmar-scaletech) [inline src/shared/constants/validation-schema.ts]: remove commented code

File: src/shared/interface/state.ts [hunk 27]
```diff
@@ -10,38 +10,38 @@ import { UserData } from 'features/login/interface/login.interface';
...
 // }
 
-export interface AuthState extends UserData {
-	isLoggedIn: boolean;
-	selectedOrg: { label: string; value: string } | null;
-}
+// export interface AuthState extends UserData {
+//   isLoggedIn: boolean;
+//   selectedOrg: { label: string; value: string } | null;
+// }
```
Comment (daxesh-parmar-scaletech) [inline src/shared/interface/state.ts]: remove commented code

File: src/shared/services/auth.service.ts [hunk 28]
```diff
@@ -16,19 +13,20 @@ const checkLogin = (): boolean => {
...
     const data = localStorage.authData;
-
     if (data) {
       const bytes = CryptoJS.AES.decrypt(data.toString(), KEY);
-      const decryptedData: UserData = JSON.parse(
+      const decryptedData = JSON.parse(
         bytes.toString(CryptoJS.enc.Utf8)
-      );
-      return decryptedData && decryptedData.token ? decryptedData.token : false;
+      ) as any;
```
Comment (daxesh-parmar-scaletech) [inline src/shared/services/auth.service.ts]: remove any

File: src/shared/services/http.service.ts [hunk 29]
```diff
@@ -120,18 +121,18 @@ const commonAxios = ({
...
   } else {
-    // headers['x-request-language'] = localStorage.getItem('lang');
+    headers["x-request-language"] = localStorage.getItem("lang");
   }
 
   // const selectedOrg = AuthService.getSelectedOrg();
   // if (selectedOrg) {
-
-  // 	headers.organizationId = selectedOrg.value;
+  //   headers.organizationId = selectedOrg.value;
```
Comment (daxesh-parmar-scaletech) [inline src/shared/services/http.service.ts]: remove commented code

**Review by daxesh-parmar-scaletech — CHANGES_REQUESTED** [review]

File: .env (hunk 1 above)
Comment (khushbuSatasiya) [reply .env]: done

File: public/index.html (hunk 2 above)
Comment (khushbuSatasiya) [reply public/index.html:26]: done

File: src/App.css (hunk 3 above)
Comment (khushbuSatasiya) [reply src/App.css]: done

File: src/App.tsx (hunk 4 above)
Comment (khushbuSatasiya) [reply src/App.tsx]: done

File: src/App.tsx (hunk 5 above)
Comment (khushbuSatasiya) [reply src/App.tsx]: because it's not necessary there.

File: src/features/User/component/user.tsx (hunk 6 above)
Comment (khushbuSatasiya) [reply src/features/User/component/user.tsx]: done

File: src/features/User/component/viewUser.tsx (hunk 7 above)
Comment (khushbuSatasiya) [reply src/features/User/component/viewUser.tsx:23]: done

File: src/features/checkIn/component/checkIn.tsx (hunk 8 above)
Comment (khushbuSatasiya) [reply src/features/checkIn/component/checkIn.tsx]: done

File: src/features/checkOut/component/addExtraTaskForm.tsx (hunk 10 above)
Comment (khushbuSatasiya) [reply src/features/checkOut/component/addExtraTaskForm.tsx]: done

File: src/features/dashboard/component/dashboard.tsx (hunk 15 above)
Comment (khushbuSatasiya) [reply src/features/dashboard/component/dashboard.tsx]: done

File: src/shared/constants/validation-schema.ts (hunk 25 above)
Comment (khushbuSatasiya) [reply src/shared/constants/validation-schema.ts]: done

File: src/shared/components/table/container/table.tsx (hunk 24 above)
Comment (khushbuSatasiya) [reply src/shared/components/table/container/table.tsx]: done

File: src/shared/constants/validation-schema.ts (hunk 26 above)
Comment (khushbuSatasiya) [reply src/shared/constants/validation-schema.ts]: done

File: src/shared/interface/state.ts (hunk 27 above)
Comment (khushbuSatasiya) [reply src/shared/interface/state.ts]: done

File: src/shared/services/http.service.ts (hunk 29 above)
Comment (khushbuSatasiya) [reply src/shared/services/http.service.ts]: done

File: src/shared/components/noRecords/noRecords.tsx (hunk 22 above)
Comment (khushbuSatasiya) [reply src/shared/components/noRecords/noRecords.tsx]: why?

Comment (daxesh-parmar-scaletech) [reply src/shared/components/noRecords/noRecords.tsx]: do you use dark theme in this application?

File: src/features/checkOut/component/checkOut.tsx (hunk 13 above)
Comment (khushbuSatasiya) [reply src/features/checkOut/component/checkOut.tsx]: done

File: src/features/checkOut/component/addExtraTaskForm.tsx (hunk 9 above)
Comment (khushbuSatasiya) [reply src/features/checkOut/component/addExtraTaskForm.tsx]: done

File: src/features/dashboard/component/dashboard.tsx (hunk 14 above)
Comment (khushbuSatasiya) [reply src/features/dashboard/component/dashboard.tsx]: done

File: src/features/dashboard/component/userDetail.tsx (hunk 16 above)
Comment (khushbuSatasiya) [reply src/features/dashboard/component/userDetail.tsx]: done

File: src/shared/components/noRecords/noRecords.tsx (hunk 22 above)
Comment (khushbuSatasiya) [reply src/shared/components/noRecords/noRecords.tsx]: no

Comment (khushbuSatasiya) [reply src/shared/components/noRecords/noRecords.tsx]: done

File: src/features/checkOut/component/checkOut.tsx (hunk 11 above)
Comment (khushbuSatasiya) [reply src/features/checkOut/component/checkOut.tsx]: done

File: src/features/checkOut/component/checkOut.tsx (hunk 12 above)
Comment (khushbuSatasiya) [reply src/features/checkOut/component/checkOut.tsx:51]: done

File: src/features/leave/component/leave.tsx (hunk 19 above)
Comment (khushbuSatasiya) [reply src/features/leave/component/leave.tsx]: done

File: src/shared/components/notification/notification.tsx (hunk 23 above)
Comment (khushbuSatasiya) [reply src/shared/components/notification/notification.tsx]: done

File: src/shared/services/auth.service.ts (hunk 28 above)
Comment (khushbuSatasiya) [reply src/shared/services/auth.service.ts]: done

File: src/features/project/components/teamCalendar.tsx (hunk 21 above)
Comment (khushbuSatasiya) [reply src/features/project/components/teamCalendar.tsx]: done

File: src/features/leave/component/leave.tsx (hunk 20 above)
Comment (khushbuSatasiya) [reply src/features/leave/component/leave.tsx]: done

File: src/features/dashboard/component/userInfoPopup.tsx (hunk 18 above)
Comment (khushbuSatasiya) [reply src/features/dashboard/component/userInfoPopup.tsx]: done

File: src/features/dashboard/component/userDetail.tsx (hunk 17 above)
Comment (khushbuSatasiya) [reply src/features/dashboard/component/userDetail.tsx]: done

**Review by daxesh-parmar-scaletech — APPROVED** [review]

**Pull request merged**